components/constituency_lookup/ConstituencyLookup.tsx

```diff
@@ -18,17 +18,35 @@ import {
   postcodeInputPattern,
   validatePostcode,
 } from "@/utils/Postcodes";
+import { submitANForm } from "@/utils/AnApiSubmission";
 import { rubik } from "@/utils/Fonts";
 import FormCheckInput from "react-bootstrap/esm/FormCheckInput";
 import FormCheckLabel from "react-bootstrap/esm/FormCheckLabel";
-import { useMemo, useRef, useState } from "react";
+import { useMemo, useRef, useState, useEffect } from "react";
 
-const errorCodeToErrorMessage = (code: ErrorCode) => {
+const postcodeErrorToErrorMessage = (code: PostCodeErrorCode) => {
   switch (code) {
     case "POSTCODE_INVALID":
-      return "Oops, that postcode doesn't look right to us. Please try again or contact us.";
+      return "Oops, that postcode doesn't look right to us. Please try again or contact us."; // TODO make contact us a link?
+
     case "POSTCODE_NOT_FOUND":
       return "Oops, we can't find that postcode. Please try again or contact us.";
+    case "UNCLEAR_CONSTITUENCY":
+      return "Please select your address or your constituency from the list.";
+    case "SERVER_ERROR":
+      return "Something went wrong looking up your constituency.";
+    //TODO add in a link for them to find their constituency in a list
+    default:
+      return "";
+  }
+};
+
+const emailErrorToErrorMessage = (code: EmailErrorCode) => {
+  switch (code) {
+    case "EMAIL_INVALID":
+      return "Please add a valid email address.";
+    case "SERVER_ERROR":
+      return "Something went wrong signing you up. Please try again?";
```

Comment:
Not for now, but probably worth thinking about whether we give people the option to go to their constituency page if the email signup form submission keeps failing (eg. Action Network is down).

You _can_ just untick the email opt-in box, but not all people will figure that out / it won't be a great user experience.

Also thinking about if/how we'll even find out if the Action Network submission is failing, since it's all client-side :thinking:

Reply:
Both good points I think it will really only fail if AN has some kind of catastrophic failure... but they probably won't tell us if that happens... Thinking about it maybe if there is a server error it should just wait a few seconds and re-try and then give up.